swarm/utils/metrics.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+//NOTE: The metrics package for swarm should align with the general go-ethereum metrics,
+//which uses the `rcrowley/go-metrics` package. The latter has limitations with
+//aggregating and resetting metrics, which sparked some research on the swarm team
+//regarding metrics. Therefore, this swarm metrics package introduces some indirection,
+//which abstracts a metrics API, allowing to use a different metrics library if needed.
+//If ultimately the go-metrics issues are solved,
+//it may be omitted in favor of directly using the `metrics/metrics` package in go-ethereum
+
+package utils
```

Comment:
My suggestion to rename this package to metrics. For example, metrics.Increment is more obvious then utils.Increment.

Reply:
The reason I have called it `utils` is that there exists a package `metrics` in `go-ethereum`, for disambiguation. In fact this same `metrics.go` file needs to use it `go-ethereum/metrics` , so it may become confusing. I have no other issue and no strong opinion on renaming this whatever we want.

Reply:
Any reason not to just put the code in the `go-ethereum/metrics` package?

Reply:
This work started as evaluation of different metrics packages. Still a few days ago we were wanting to use the `statsd` client and go an independent path from `go-ethereum`. For this reason I introduced this package as an abstraction and a "hedge". It today seems we have a final decision to stick to `go-metrics` and `go-ethereum`, which would make this abstraction layer obsolete.

This decision would require a rewrite of all metrics calls in this PR to use `go-ethereum` metrics. Also, there is still the code needed to send metrics via Graphite protocol to influxdb which isn't inside `go-ethereum/metrics`. Would that be moved over there as well?

Reply:
@janos @nonsense @lmars this would require a quick decision here

Reply:
I will try to get the PR with the fork of `go-metrics` done asap. I am thinking of including the InfluxDB reporter inside for simplicity. In this case the abstraction would be indeed obsolete. Will keep you posted on progress.

Reply:
I guess the reason @holisticode created this package is mostly to differentiate swarm metrics from geth metrics, and to have an abstraction layer when we were still deciding between go-metrics and statsd which have different clients.

However now that we have come to a conclusion that we will be using go-metrics, this abstraction layer is no longer necessary. Also I think we should be using the `DefaultRegistry`, rather than the `SwarmRegistry`, as there will most probably be common code that we want to measure from `go-ethereum` within `swarm` (such as the `p2p` package). If we had 2 registries, we'd also need 2 reporters, etc. - something that seems like an overkill.

So ultimately I suggest we get rid of the `utils` package, and go directly with the go-ethereum/metrics package.

Reply:
At this point, I agree